regulation-worker/internal/delete/delete.go

```diff
@@ -3,31 +3,44 @@ package delete
 import (
 	"context"
 	"fmt"
+	"sync"
 
 	"github.com/rudderlabs/rudder-server/regulation-worker/internal/model"
 )
 
-type deleter interface {
-	Delete(ctx context.Context, job model.Job, dest model.Destination) (model.JobStatus, error)
+//go:generate mockgen -source=delete.go -destination=mock_delete_test.go -package=delete github.com/rudderlabs/rudder-server/regulation-worker/internal/delete
+type deleteManager interface {
+	Delete(ctx context.Context, job model.Job, destConfig map[string]interface{}, destName string) model.JobStatus
+	GetSupportedDestinations() []string
 }
-type Deleter struct {
-	API   deleter
-	Batch deleter
+
+type DeleteRouter struct {
+	Managers []deleteManager
+	router   map[string]deleteManager
+	once     sync.Once
 }
 
-//get destType & access credentials from workspaceID & destID
-//call appropriate struct file type or api type based on destType.
-func (d *Deleter) DeleteJob(ctx context.Context, job model.Job, dest model.Destination) (model.JobStatus, error) {
-	switch dest.Type {
-	case "api":
-		delAPI := MockAPIDeleter{}
-		return delAPI.Delete(ctx, job, dest)
-	case "batch":
-		delBatch := MockBatchDeleter{}
-		return delBatch.Delete(ctx, job, dest)
-	default:
-		fmt.Println("default called")
-		return model.JobStatusFailed, fmt.Errorf("deletion feature not available for %s destination type", dest.Type)
+func New(managers ...deleteManager) *DeleteRouter {
```

Comment:
`delete.New` -> `delete.NewRouter` to remove ambiguity 

```suggestion
func NewRouter(managers ...deleteManager) *DeleteRouter {
```